truffleruby/src/main/ruby/core/rbconfig.rb

```diff
@@ -78,50 +77,124 @@ module RbConfig
+      'LIBEXT' => 'c',
+      'OBJEXT' => 'bc',
+      'exeext' => '',
+      'EXEEXT' => '',
+      'EXECUTABLE_EXTS' => '',
+      'includedir' => '',
+      'LIBS' => libs,
+      'DLDLIBS' => '',
+      'LIBRUBYARG_STATIC' => '',
+      'LIBRUBYARG_SHARED' => '',
+      'libdirname' => 'libdir',
+      'LIBRUBY' => '',
+      'LIBRUBY_A' => '',
+      'LIBRUBYARG' => '',
+      'prefix' => '',
+      'ruby_install_name' => ruby_install_name,
+      'RUBY_INSTALL_NAME' => ruby_install_name,
+      'ruby_version' => ruby_api_version,
+      'RUBY_BASE_NAME' => ruby_base_name,
+      'RUBY_SO_NAME' => ruby_so_name
+  }
+
   ruby_home = Truffle::Boot.ruby_home
 
   if ruby_home
     libdir = "#{ruby_home}/lib"
     bindir = "#{libdir}/bin"
+    prefix = ruby_home
+    exec_prefix = prefix
+    libdir =  "#{exec_prefix}/lib"
+    rubylibprefix = "#{libdir}/#{ruby_base_name}"
+    rubylibdir = "#{rubylibprefix}/#{ruby_version}"
+    rubyarchdir = "#{rubylibdir}/#{arch}"
+    archdir = rubyarchdir
+    sitearch = arch
+    sitedir = "#{rubylibprefix}/site_ruby"
+    sitelibdir = "#{sitedir}/#{ruby_version}"
+    sitearchdir = "#{sitelibdir}/#{sitearch}"
+    topdir = archdir
 
     CONFIG.merge!({
+      'prefix'   => prefix,
+      'exec_prefix' => exec_prefix,
+      'libdir'       =>  libdir,
+      'rubylibprefix' => rubylibprefix,
+      'rubylibdir' => rubylibdir,
+      'rubyarchdir' => rubyarchdir,
+      'archdir' => archdir,
       'bindir' => bindir,
-      'libdir' => libdir,
-      "sitelibdir"=>"#{ruby_home}/lib/2.3/site_ruby", # TODO BJF Oct 21, 2016 Need to review these values
-      "sitearchdir"=>"#{ruby_home}/lib/2.3/site_ruby",
+      'hdrdir' => "#{ruby_home}/lib/cext",
+      'sitearch' => sitearch,
+      'sitedir' => sitedir,
+      'sitelibdir' => sitelibdir,
+      'sitearchdir' => sitearchdir,
       'rubyhdrdir' => "#{libdir}/cext",
-      'topdir' => "#{ruby_home}/lib/stdlib", # TODO CS 21-Jan-17 this doesn't exist any more
-      "rubyarchhdrdir"=>"#{libdir}/cext",
+      'topdir' => topdir,
+      'rubyarchhdrdir' =>"#{libdir}/cext",
     })
-
     MAKEFILE_CONFIG.merge!({
-      'hdrdir' => "#{ruby_home}/lib/cext",
-      'bindir' => bindir
-    })
+          'prefix'   => ruby_home,
+          'exec_prefix' => '$(prefix)',
+          'libdir'       =>  '$(exec_prefix)/lib',
+          'rubylibprefix' => '$(libdir)/$(RUBY_BASE_NAME)',
+          'rubylibdir' => '$(rubylibprefix)/$(ruby_version)',
+          'rubyarchdir' => '$(rubylibdir)/$(arch)',
+          'archdir' => '$(rubyarchdir)',
+          'bindir' => bindir,
+          'hdrdir' => "#{ruby_home}/lib/cext",
+          'sitearch' => '$(arch)',
+          'sitedir' => '$(rubylibprefix)/site_ruby',
+          'sitelibdir' => '$(sitedir)/$(ruby_version)',
+          'sitearchdir' => '$(sitelibdir)/$(sitearch)',
+          'rubyhdrdir' => "#{libdir}/cext",
+          'topdir' => '$(archdir)',
+          'rubyarchhdrdir' =>"#{libdir}/cext",
```

Comment:
In instances where it's a single replacement:
`'RUBY_SO_NAME' => '$(RUBY_BASE_NAME)'`
Instead of
`ruby_so_name = "#{ruby_base_name}"`
I opted to use:
`ruby_so_name = ruby_base_name`

I can't say with certainty that MAKEFILE_CONFIG will always be static values. I believe the MRI RbConfig values are generated from autoconf's config.status file when ruby is installed. So, if this can vary platform to platform, there may be a need to make this dynamic in the future.

The `rubyarchhdrdir` I think I just left alone since this is a first pass at cleaning this up and I don't want to regress/affect the c extension work going on.